cookbook/add-tabs.rst

```diff
@@ -0,0 +1,300 @@
+    {
+        /**
+        * name of structure extension.
+        */
+        const SOCIAL_EXTENSION_NAME = 'social';
+
+        protected $properties = [
+            'twitter_title',
+            'twitter_description',
+            'twitter_image',
+        ];
+
+        protected $name = self::SOCIAL_EXTENSION_NAME;
+
+        protected $additionalPrefix = 'social';
+
+        public function save(NodeInterface $node, $data, $webspaceKey, $languageCode)
+        {
+            $this->setLanguageCode($languageCode, 'i18n', null);
+
+            $data = $this->encodeImages($data);
+
+            $this->saveProperty($node, $data, 'twitter_title');
+            $this->saveProperty($node, $data, 'twitter_description');
+            $this->saveProperty($node, $data, 'twitter_image');
+        }
+
+        public function load(NodeInterface $node, $webspaceKey, $languageCode)
+        {
+            $twitterImageNode = $this->loadProperty($node, 'twitter_image');
+            $twitterImage = null;
+            if ($twitterImageNode) {
+                $twitterImage = json_decode($twitterImageNode, true);
+            }
+
+            return [
+                'twitter_title' => $this->loadProperty($node, 'twitter_title'),
+                'twitter_description' => $this->loadProperty($node, 'twitter_description'),
+                'twitter_image' => $twitterImage,
+            ];
+        }
+
+        public function export($properties, $format = null)
+        {
+            $data = [];
+            foreach ($properties as $key => $property) {
+                $value = $property;
+                if (\is_bool($value)) {
+                    $value = (int) $value;
+                }
+
+                $data[$key] = [
+                    'name' => $key,
+                    'value' => $value,
+                    'type' => '',
+                ];
+            }
+
+            return $data;
+        }
+
+        public function import(NodeInterface $node, $data, $webspaceKey, $languageCode, $format)
+        {
+            $this->setLanguageCode($languageCode, 'i18n', null);
+
+            $this->save($node, $data, $webspaceKey, $languageCode);
+        }
+
+        public function getImportPropertyNames()
+        {
+            return $this->properties;
+        }
+
+        private function encodeImages(array $data)
+        {
+            if ($data['twitter_image']) {
+                $data['twitter_image'] = json_encode($data['twitter_image']);
+            }
+
+            return $data;
+        }
+    }
+
+Also this should be registered as a service with the tag `sulu.structure.extension`.
+
+.. code-block :: yaml
+
+    my_own.extension.social:
+        class: App\Structure\SocialStructureExtension
+        tags: { name: 'sulu.structure.extension' }
+
+Because we used slashes in the property names in the form. Our data is automaticly mapped to the extension data given in the view. 
+
+.. figure:: ../img/socials-tab-dump.png
```

Comment:
Wouldn't add that as a screenshot... Maybe we can add a table showing how to access which value. The thing is that the screenshot is not searchable.

Reply:
I removed the screenshot. Not sure what you mean by the table.